protocols/raft/src/main/java/io/atomix/protocols/raft/storage/snapshot/SnapshotStoreFactory.java

```diff
@@ -0,0 +1,8 @@
+package io.atomix.protocols.raft.storage.snapshot;
+
+import io.atomix.protocols.raft.storage.RaftStorage;
+
+@FunctionalInterface
+public interface SnapshotStoreFactory {
+  SnapshotStore createSnapshotStore(RaftStorage storage);
```

Comment:
Probably not, but I just extracted the interface. I'd rather focus with these PRs on extracting the existing API so that we can use it, and then we can refactor all this stuff to make more sense.